src/dali_executor/CMakeLists.txt

```
@@ -30,25 +30,27 @@ include(${tritondalibackend_SOURCE_DIR}/cmake/dali.cmake)
 
 separate_arguments(EXTRA_DOWNLOAD_ARGS UNIX_COMMAND ${DALI_DOWNLOAD_EXTRA_OPTIONS})
 
-add_custom_command(  # Download and unpack DALI wheel
-    OUTPUT dali_whl
-    COMMAND
-        pip download ${EXTRA_DOWNLOAD_ARGS} --extra-index-url ${DALI_EXTRA_INDEX_URL}
-          -d ${CMAKE_CURRENT_BINARY_DIR}
-          ${DALI_DOWNLOAD_PKG_NAME}$<$<BOOL:${DALI_VERSION}>:==${DALI_VERSION}>
+if(EXTRA_DOWNLOAD_ARGS)
+    configure_file(${tritondalibackend_SOURCE_DIR}/cmake/dalienv-extra_args.yml.in dalienv.yml @ONLY)
+else()
+    configure_file(${tritondalibackend_SOURCE_DIR}/cmake/dalienv.yml.in dalienv.yml @ONLY)
+endif()
+
+add_custom_command(
+    OUTPUT dali_env
     COMMAND
-        unzip -q -o ${CMAKE_CURRENT_BINARY_DIR}/nvidia_dali*.whl -d ${CMAKE_CURRENT_BINARY_DIR}/dali
-    COMMAND touch dali_whl  # So that this command won't be re-run every time
-    COMMENT "Acquiring DALI release"
+        conda env create -f dalienv.yml -p ${CMAKE_CURRENT_BINARY_DIR}/dalienv -q
```

I wonder if we need this additional level of indirection through dalienv.yml, how about:
```suggestion
        conda create -n dalienv python=3.8 -y && conda install -n dalienv -y pip && conda run -n dalienv pip install --extra-index-url ${DALI_EXTRA_INDEX_URL} ${EXTRA_DOWNLOAD_ARGS} ${DALI_DOWNLOAD_PKG_NAME}
```

---

I believe both would work equally well, both have pros and cons. @prak-nv , what your opinion on this?

---

@szalpal version seems to be more readable/explicit for me.

---

I'm fine with both.